src/component/Form/AccountForm.tsx

```diff
@@ -0,0 +1,230 @@
+import { Dispatch, SetStateAction, useCallback, useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { User } from "../../types/User";
+import { useTranslation } from "react-i18next";
+import { OptionsSelect } from "../../types/OptionsSelect";
+import Countries from "../../hook/Countries";
+import regex from '../../services/regex';
+import { updateUser } from "../../services/user.service";
+import { useAuth } from "../../contexts/AuthContext";
+import { toast } from "react-toastify";
+
+type IAccountForm = {
+  user: User | undefined;
+  setUser: Dispatch<SetStateAction<User | undefined>>;
+}
+
+export default function AccountFom(props: IAccountForm) {
+    const { t } = useTranslation('global');
+    const { register, handleSubmit, setValue, formState: { errors } } = useForm();
+    const { token } = useAuth();
+    const [init, setInit] = useState(true);
```

Comment:
la convention pour les booleans est de mettre un is ou has devant.
isInit dans ce cas